node/src/chain_spec.rs

```diff
@@ -24,9 +25,60 @@ use sc_chain_spec::{ChainSpecExtension, ChainSpecGroup};
 use sc_service::ChainType;
 use sc_telemetry::TelemetryEndpoints;
 use serde::{Deserialize, Serialize};
-use sp_runtime::Perbill;
+use sha3::{Digest, Keccak256};
+
+use sp_core::{ecdsa, Pair, Public, H160, H256};
+use sp_runtime::{
+	traits::{BlakeTwo256, Hash},
+	Perbill,
+};
 use stake::{InflationInfo, Range};
+use std::convert::TryInto;
 use std::{collections::BTreeMap, str::FromStr};
+use tiny_hderive::bip32::ExtendedPrivKey;
+
+/// Helper function to derive a num_accounts child pairs from mnemonics
+/// Substrate derive function cannot be used because the derivation is different than Ethereum's
+pub fn derive_pairs_from_mnemonic<TPublic: Public>(
+	mnemonic: &str,
+	num_accounts: u32,
+) -> Vec<TPublic::Pair> {
+	let seed = Mnemonic::from_phrase(mnemonic, Language::English)
+		.map(|x| Seed::new(&x, ""))
+		.expect("Wrong mnemonic provided");
+
+	let mut childs = Vec::new();
+	for i in 0..num_accounts {
+		if let Some(child_pair) =
+			ExtendedPrivKey::derive(seed.as_bytes(), format!("m/44'/60'/0'/0/{}", i).as_ref())
+				.ok()
+				.map(|account| TPublic::Pair::from_seed_slice(&account.secret()).ok())
+				.flatten()
+		{
+			childs.push(child_pair);
+		} else {
+			log::error!("An error ocurred while deriving key {} from parent", i)
+		}
+	}
+	childs
+}
+
+/// Helper function to get an AccountId from Key Pair
+/// We need the full decompressed public key to derive an ethereum-style account
+/// Substrate does not provide a method to obtain the full decompressed public key
+/// Therefore, this function uses the secp256k1_ecdsa_recover method to recover the full key
+pub fn get_account_id_from_pair<TPublic: Public>(pair: TPublic::Pair) -> Option<AccountId> {
```

Comment:
It feels weird that this requires a private key. I get that it's because this trick requires signing a message, and I'm fine keeping it this way as a stopgap. But we should at least comment how it might be improved.